src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/swagger-x-ms-discriminator-value.json

```diff
@@ -0,0 +1,287 @@
+            "items": {
+              "type": "string"
+            },
+            "collectionFormat": "csv"
+          },
+          {
+            "name": "limit",
+            "in": "query",
+            "description": "maximum number of results to return",
+            "required": false,
+            "type": "integer",
+            "format": "int32"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "pet response",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/pet"
+              }
+            }
+          },
+          "default": {
+            "description": "unexpected error",
+            "schema": {
+              "$ref": "#/definitions/errorModel"
+            }
+          }
+        }
+      },
+      "post": {
+        "description": "Creates a new pet in the store.  Duplicates are allowed",
+        "operationId": "addPet",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "pet",
+            "in": "body",
+            "description": "Pet to add to the store",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/newPet"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "pet response",
+            "schema": {
+              "$ref": "#/definitions/pet"
+            }
+          },
+          "default": {
+            "description": "unexpected error",
+            "schema": {
+              "$ref": "#/definitions/errorModel"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [],
+        "summary": "",
+        "description": "",
+        "operationId": "CreateOrUpdatePet",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "PetCreateOrUpdateParameter",
+            "in": "body",
+            "description": "A Pet",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/pet"
+            }
+          }
+        ],
+        "responses": {
+          "200": { "$ref": "#/responses/petResponse" },
+          "default": {
+            "description": "unexpected error",
+            "schema": {
+              "$ref": "#/definitions/errorModel"
+            }
+          }
+        },
+        "schemes": [ "http", "https" ],
+        "deprecated": true,
+        "security": [],
+        "x-test-header-info": {
+          "name": "x-static-header",
+          "value": "headerValue"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "pet": {
+      "required": [
+        "id",
+        "name",
+        "type"
+      ],
+      "discriminator": "type",
+      "x-ms-discriminator-value": "Microsoft.Models.MSPet",
```

Comment:
@lmazuel - Base model "pet" has the discriminator and the x-ms-discriminator-value extension.